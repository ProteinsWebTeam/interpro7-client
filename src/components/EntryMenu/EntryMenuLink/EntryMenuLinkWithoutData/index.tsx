import React from 'react';

import Link from 'components/generic/Link';
import NumberComponent from 'components/NumberComponent';

import cssBinder from 'styles/cssBinder';

import pages from 'pages/style.css';
import fonts from 'EBI-Icon-fonts/fonts.css';
import local from '../style.css';

const css = cssBinder(pages, fonts, local);

const icons = new Map([
  ['Overview', { icon: '\uF2BB', class: 'icon-common' }],
  ['Entries', { icon: '\uF1C0', class: 'icon-common' }],
  ['Proteins', { icon: 'P', class: 'icon-conceptual' }],
  ['Structures', { icon: 's', class: 'icon-conceptual' }],
  ['Domain Architectures', { icon: undefined, class: 'icon-count-ida' }],
  ['Taxonomy', { icon: undefined, class: 'icon-count-species' }],
  ['Proteomes', { icon: undefined, class: 'icon-count-proteome' }],
  ['Sets', { icon: undefined, class: 'icon-count-set' }],
  ['Signature', { icon: undefined, class: 'icon-count-hmm' }],
  ['Alignments', { icon: '\uF1DE', class: 'icon-common' }],
  ['Sequence', { icon: '\uF120', class: 'icon-common' }],
  ['Curation', { icon: undefined, class: 'icon-common' }],
  ['New Structural Model', { icon: undefined, class: 'icon-common' }],
]);

type Props = {
  name: string;
  loading: boolean;
  to:
    | InterProPartialLocation
    | ((location: InterProPartialLocation) => InterProPartialLocation);
  value?: number;
  exact?: boolean;
  collapsed?: boolean;
  usedOnTheSide?: boolean;
};
export const EntryMenuLinkWithoutData = ({
  name,
  value,
  loading,
  to,
  exact,
  usedOnTheSide,
  collapsed,
}: Props) => {
  const hasValue = value !== null && value !== undefined;
  return (
    <li
      className={css('tabs-title', {
        ['used-on-the-side']: usedOnTheSide,
        collapsed,
      })}
      data-testid={`menu-${name.toLowerCase().replace(/\s+/g, '_')}`}
    >
      <Link
        to={to}
        exact={exact}
        className={css('browse-tabs-link', {
          'withuot-counter': !hasValue || isNaN(value as unknown as number),
        })}
        activeClass={css('is-active', 'is-active-tab')}
      >
        <span data-content={name} className={css('name')}>
          <i
            data-icon={(icons.get(name) || {}).icon}
            className={css(
              'icon',
              (icons.get(name) || {}).class,
              'icon-count-sm',
              'margin-right-medium',
            )}
            aria-label={`icon ${name}`}
            data-testid={`entry-menu-${name
              .toLowerCase()
              .replace(/\s+/g, '_')}`}
          />
          {name}
        </span>
        {hasValue && ' '}
        {hasValue && !isNaN(value) && (
          <NumberComponent
            label
            loading={loading}
            abbr
            duration={usedOnTheSide ? 0 : undefined}
            className={css('counter')}
            noAnimation
          >
            {value}
          </NumberComponent>
        )}
      </Link>
    </li>
  );
};

export default EntryMenuLinkWithoutData;
