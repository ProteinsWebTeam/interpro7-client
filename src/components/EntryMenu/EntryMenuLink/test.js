import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { EntryMenuLink } from '.';
import EntryMenuLinkWithoutData from './EntryMenuLinkWithoutData';

const renderer = new ShallowRenderer();

describe('<EntryMenuLinkWithoutData />', () => {
  test('should render', () => {
    renderer.render(
      <EntryMenuLinkWithoutData
        name={'Structures'}
        value={41369}
        to={() => {}}
        loading={false}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('<EntryMenuLink />', () => {
  test('should render', () => {
    renderer.render(
      <EntryMenuLink
        to={() => {}}
        name={'Proteins'}
        counter={'proteins'}
        data={{
          loading: false,
          payload: {
            metadata: {
              accession: '9606',
              counters: {
                dbEntries: {
                  cathgene3d: 2233,
                  cdd: 5912,
                  hamap: 389,
                  interpro: 16561,
                  mobidblt: 1,
                  panther: 24795,
                  pfam: 6622,
                  pirsf: 960,
                  prints: 1488,
                  profile: 869,
                  prosite: 884,
                  sfld: 64,
                  smart: 1024,
                  ssf: 1129,
                  tigrfams: 595,
                },
                domain_architectures: 13877,
                entries: 63526,
                proteins: 171145,
                proteomes: 1,
                sets: 5750,
                structures: 41369,
              },
              lineage:
                ' 1 131567 2759 33154 33208 6072 33213 33511 7711 89593 7742 7776 117570 117571 8287 1338369 32523 32524 40674 32525 9347 1437010 314146 9443 376913 314293 9526 314295 9604 207598 9605 9606 ',
              name: {
                name: 'Homo sapiens',
                short: 'Homo sapiens (Human)',
              },
              parent: '9605',
              rank: 'species',
              source_database: 'uniprot',
            },
          },
        }}
        mainKey={'taxonomy'}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
