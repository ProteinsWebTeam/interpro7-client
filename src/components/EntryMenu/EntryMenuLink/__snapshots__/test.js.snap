// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryMenuLink /> should render 1`] = `
<EntryMenuLinkWithoutData
  loading={false}
  name="Proteins"
  to={[Function]}
  value={171145}
/>
`;

exports[`<EntryMenuLinkWithoutData /> should render 1`] = `
<li
  className="tabs-title"
  data-testid="menu-structures"
>
  <Memo(Connect(Link))
    activeClass="is-active is-active-tab"
    className="browse-tabs-link"
    to={[Function]}
  >
    <span
      className="name"
      data-content="Structures"
    >
      <i
        aria-label="icon Structures"
        className="icon icon-conceptual icon-count-sm margin-right-medium"
        data-icon="s"
        data-testid="entry-menu-structures"
      />
      Structures
    </span>
     
    <Memo(Connect(NumberComponent))
      abbr={true}
      className="counter"
      label={true}
      loading={false}
    >
      41369
    </Memo(Connect(NumberComponent))>
  </Memo(Connect(Link))>
</li>
`;
